lib/LANraragi/Plugin/Metadata/CopyArchiveTags.pm

```diff
@@ -65,16 +47,16 @@ sub internal_get_tags {
 
     $logger->info("Copying tags from archive \"${lrr_gid}\"");
 
-    my $tags = LANraragi::Utils::Database::get_tags($lrr_gid);
+    my $tags = LANraragi::Utils::Database::get_tags($lrr_gid) || '';
 
     if ( !$params->{'copy_date_added'} ) {
         my @tags = split_tags_to_array($tags);
         $tags = join_tags_to_string( grep( !m/date_added/, @tags ) );
     }
 
-    my %hashdata = ( tags => $tags );
+    $logger->info( "Sending the following tags to LRR: " . ( $tags || '-' ) );
 
-    return %hashdata;
+    return ( tags => $tags || '' );
```

Comment:
I was wondering why not returning the tags as an array? It should reduce the number of splits an joins around the project.

Reply:
Metadata plugin return tags are just string-concatenated to the existing tags, so this is technically faster than doing a split/join dance server-wise.  

Although this was written before the introduction of search indexes where we still need to do splits regardless.. Since this is front-facing behavior used by plugin devs I wouldn't try changing it now unfortunately 🤷  

(also since most metadata sources just spit out tags as a comma-separated string anyway I think it usually makes their lives easier)